### ISSUE TITLE
TorchModel hangs before training 2nd epoch

### ISSUE THREAD

**Issue by alpha-tango** [vskode/acodet#47]
I tried running the TorchModel on an h200 with the full set of training data, and it trained 1 epoch successfully, but 20 minutes later, hadn't started the next epoch. When I killed the job, the traceback was:

```
    run_training(**kwargs)
  File "/gpfs2/scratch/atpope/bioacoustics/acodet/acodet/train.py", line 300, in run_training
    model = train(model, data_loaders)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[my_folder]/acodet/acodet/torch_train.py", line 115, in train
    for batch in val_loader:
  File "[my_folder]/acodet/.venv/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 741, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "[my_folder]/acodet/.venv/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1524, in _next_data
    idx, data = self._get_data()
                ^^^^^^^^^^^^^^^^
  File "[my_folder]/acodet/.venv/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1473, in _get_data
    success, data = self._try_get_data()
                    ^^^^^^^^^^^^^^^^^^^^
  File "[my_folder]/acodet/.venv/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 1310, in _try_get_data
    data = self._data_queue.get(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[user]/.local/share/uv/python/cpython-3.11.14-linux-x86_64-gnu/lib/python3.11/queue.py", line 180, in get
    self.not_empty.wait(remaining)
  File "[user]/.local/share/uv/python/cpython-3.11.14-linux-x86_64-gnu/lib/python3.11/threading.py", line 331, in wait
    gotit = waiter.acquire(True, timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyboardInterrupt
```

**Comment by vskode** [comment]
ok the problem was that the entire validation set was loaded, which took ages. now only a validation set corresponding to the steps_per_epoch - meaning to the size of the train set per epoch, is loaded.
